code/include/swoc/DiscreteRange.h

```diff
@@ -828,6 +829,10 @@ template <typename METRIC, typename PAYLOAD> class DiscreteSpace {
 
   ~DiscreteSpace();
 
+  void mark_bulk(std::vector<std::pair<range_type, PAYLOAD>> const &marks);
+
+  void mark_bulk(std::pair<range_type, PAYLOAD>* start, size_t n);
```

Comment:
doxygen comments woud be helpful (comment also applies to other new parameters, like updateTree).